### DISCUSSION TITLE
初始值 setInitialValues 时，会触发 onFieldChange，导致无法正确初始值。

### DISCUSSION THREAD

**Discussion by Z8264** [alibaba/formily#2530] [Q&A 2.x] [answered]
请教下，这个问题怎么解决？

**Comment by janryWang** [comment, marked as answer]
具体是啥问题，给个codesandbox看看

**Reply by Z8264** [reply]
https://codesandbox.io/s/lively-cookies-hngyj?file=/src/App.js

我设置了联动，在更换radio的时候，更换select的数据集，并将selet的值设置为空。
但是，我初始化数据的时候需要设置默认值setInitialValues，触发了联动监听，导致数据为空。

**Reply by Z8264** [reply]
看一下这个场景我该如何解决。

**Reply by janryWang** [reply]
建议先看 https://formilyjs.org/zh-CN/guide/learn-formily

回到问题上，这个文档都说的很清楚的，以后尽量少提文档搬运的问题 https://core.formilyjs.org/zh-CN/api/entry/field-effect-hooks#onfieldinputvaluechange
